firewood/src/merkle/node.rs

```diff
@@ -375,8 +384,27 @@ impl Storable for Node {
             None
         };
 
+        let mut start_index = TRIE_HASH_LEN + 1;
+        let end_index = start_index + size_of::<u64>();
+        #[allow(clippy::indexing_slicing)]
+        let encoded_len = u64::from_le_bytes(
+            meta_raw.as_deref()[start_index..end_index]
+                .try_into()
+                .expect("invalid slice"),
+        );
```

Comment:
Avoid indexing_slicing for new code. There are three ways around this that I can think of, possibly more:
```rust
        let encoded_len = u64::from_le_bytes(
            meta_raw.as_deref().get(start_index..end_index)
                .expect("reason why meta_raw must be at least end_length bytes goes here")
                .try_into()
                .expect("invalid slice"),
        );
```

Another fix would be to use an iterator, with a minus of a memory allocation here (you'd need to delete end_index as well):
```rust
        let encoded_len = u64::from_le_bytes(
            meta_raw
                .as_deref()
                .into_iter()
                .skip(start_index)
                .take(size_of::<u64>())
                .collect::<Vec<_>>()
                .try_into()
                .expect("why we know start_index and end_index are valid offsets goes here"),
        );
```
The best fix for this is to use the bytemuck crate. We're converting a `&[u8]` to a `u64` which is what bytemuck does efficiently and safely:

```rust
        let encoded_len = u64::from_le_bytes(
            *bytemuck::from_bytes::<[u8; size_of::<u64>()]>(meta_raw
                .as_deref()
                .get(start_index..end_index)
                .expect("why we know start_index and end_index are valid offsets goes here"),
                )
        );
```

Reply:
>The best fix for this is to use the bytemuck crate.

I think this is true, but I think you want to use the [`try_cast`](https://docs.rs/bytemuck/latest/bytemuck/fn.try_cast.html) function with `Meta` and go straight from `meta_raw` to a `Meta` struct. You need to use the macros to implement the proper traits for the `Meta` struct. This was always my intention, but I just didn't get there.